### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by arb**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+sudo: false
+language: node_js
+node_js:
+  - "4"
+  - "5"
```
Comment (cjihrig) [inline .travis.yml]: 6?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,104 @@
 # white-out
-Transform stream to censor object data.
+
+A transform stream used to censor data from objects before passing them down the pipeline.
```
Comment (cjihrig) [inline README.md]: I thought we traditionally did all the badges right after the module name. Then started talking about it.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,104 @@
...
+[![Current Version](https://img.shields.io/npm/v/white-out.svg)](https://www.npmjs.org/package/white-out)
+[![Travis CI](https://travis-ci.org/continuationlabs/white-out.svg?branch=master)](https://travis-ci.org/continuationlabs/white-out)
+
+[![belly-button-style](https://cdn.rawgit.com/continuationlabs/belly-button/master/badge.svg)](https://github.com/continuationlabs/belly-button)
+
+## `new WhiteOut (filter, [options])`
+
+Creates a new `WhiteOut` transform stream with the following arguments.
+- `filter` - a key value pair where `key` is the property value to censor on the object payload and the `value` is the censor type.
+  - Valid options for `value` are "censor", "remove" and a RegExp.
```
Comment (cjihrig) [inline README.md]: Why not put "censor" and "remove" in backticks.

Comment (cjihrig) [inline README.md]: Nevermind. they are string values, not field names.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
+{
+  "name": "white-out",
+  "version": "0.0.1",
```
Comment (cjihrig) [inline package.json]: I think this should start at 0.1.0

File: package.json [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    "good"
+  ],
+  "author": "Continuation Labs <contact@continuation.io> (http://continuation.io/)",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/continuationlabs/white-out/issues"
+  },
+  "homepage": "https://github.com/continuationlabs/white-out",
+  "devDependencies": {
+    "belly-button": "^2.0.1",
```
Comment (cjihrig) [inline package.json]: `2.x.x`

File: index.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    return obj[key];
+  }
+};
+const replacer = (match, group) => {
+  return 'X'.repeat(match.length);
+};
+
+class WhiteOut extends Stream.Transform {
+  constructor (rules, options) {
+    options = options || {};
```
Comment (cjihrig) [inline index.js]: You could `Object.assign()` this to remove the branch point.

**Comment by cjihrig** [comment]
Probably don't need it at this point, but it would be nice to be able to pass a function. Or, to pass an object to allow something like censor to be configured to use something other than X. Mostly LGTM though.

File: .travis.yml (hunk 1 above)
Comment (arb) [inline .travis.yml]: I mean we used to have a standard CI setup for CL... this is a deviation from that already but I copied what you had for toolbag.

File: package.json (hunk 4 above)
Comment (arb) [inline package.json]: I was going to version 1.0.0 after this merge.

File: .travis.yml (hunk 1 above)
Comment (cjihrig) [inline .travis.yml]: 6 was just released like an hour ago :-)

And we still do have a setup. We still use it in a lot of modules - https://github.com/continuationlabs/cl.ci/blob/master/js.travis.yml

We can probably drop everything before 4 at this point.

**Pull request merged**